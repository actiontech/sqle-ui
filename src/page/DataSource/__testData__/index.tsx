export const dataSourceList = [
  {
    instance_name: 'db1',
    db_host: '20.20.20.2',
    db_port: '3306',
    db_user: 'root',
    desc: '',
    workflow_template_name: 'default',
  },
];

export const dataSourceInstance = {
  instance_name: 'db1',
  db_host: '20.20.20.2',
  db_port: '3306',
  db_user: 'root',
  db_type: 'mysql',
  desc: '',
  workflow_template_name: 'workflow-template-name-1',
};

export const dataSourceMetas = [
  {
    db_type: 'mysql',
    params: [
      { description: '字段a', name: 'a', type: 'string', value: '123' },
      { description: '字段b', name: 'b', type: 'int', value: '123' },
      { description: '字段c', name: 'c', type: 'bool', value: 'true' },
    ],
  },
  {
    db_type: 'oracle',
    params: [
      { description: '字段e', name: 'e', type: 'string', value: '2222' },
      { description: '字段f', name: 'f', type: 'int', value: '333' },
      { description: '字段mm', name: 'mm', type: 'bool', value: 'false' },
    ],
  },
];
