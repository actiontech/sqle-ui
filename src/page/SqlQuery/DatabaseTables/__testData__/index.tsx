export const tablesName = [
  { name: 'audit_plan_report_sqls_v2' },
  { name: 'audit_plan_reports_v2' },
  { name: 'audit_plan_sqls_v2' },
  { name: 'audit_plans' },
  { name: 'execute_sql_detail' },
  { name: 'global_configuration_ldap' },
  { name: 'global_configuration_license' },
  { name: 'global_configuration_oauth2' },
  { name: 'global_configuration_smtp' },
  { name: 'global_configuration_wechat' },
  { name: 'instance_role' },
  { name: 'instance_rule_template' },
  { name: 'instances' },
  { name: 'role_operations' },
  { name: 'roles' },
  { name: 'rollback_sql_detail' },
  { name: 'rule_template_rule' },
  { name: 'rule_templates' },
  { name: 'rules' },
  { name: 'sql_query_execution_sqls' },
  { name: 'sql_query_histories' },
  { name: 'sql_whitelist' },
  { name: 'system_variables' },
  { name: 'tasks' },
  { name: 'user_group_roles' },
  { name: 'user_group_users' },
  { name: 'user_groups' },
  { name: 'user_role' },
  { name: 'users' },
  { name: 'workflow_record_history' },
  { name: 'workflow_records' },
  { name: 'workflow_step_template_user' },
  { name: 'workflow_step_templates' },
  { name: 'workflow_step_user' },
  { name: 'workflow_steps' },
  { name: 'workflow_templates' },
  { name: 'workflows' },
];
