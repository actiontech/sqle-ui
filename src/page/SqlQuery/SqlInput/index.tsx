import { useBoolean } from 'ahooks';
import { Button, Card, Col, Form, InputNumber, Row, Select, Space } from 'antd';
import { useCallback, useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import MonacoEditor from 'react-monaco-editor';
import instance from '../../../api/instance';
import sql_query from '../../../api/sql_query';
import {
  IGetSQLResultParams,
  IPrepareSQLQueryParams,
} from '../../../api/sql_query/index.d';
import {
  FilterFormColLayout,
  FilterFormRowLayout,
  ResponseCode,
} from '../../../data/common';
import useChangeTheme from '../../../hooks/useChangeTheme';
import useInstance from '../../../hooks/useInstance';
import useInstanceSchema from '../../../hooks/useInstanceSchema';
import useMonacoEditor from '../../../hooks/useMonacoEditor';
import useStyles from '../../../theme';
import { ISqlInputForm, SqlQueryResultType } from '../index.type';
import { SqlInputProps } from './index.type';
import SqlQueryHistory from './Modal/SqlQueryHistory';

const defaultMaxQueryRows = 100;

const SqlInput: React.FC<SqlInputProps> = ({
  form,
  setQueryRes,
  setResultErrorMessage,
}) => {
  const { t } = useTranslation();
  const theme = useStyles();
  const { currentEditorTheme } = useChangeTheme();
  const { editorDidMount } = useMonacoEditor(form, { formName: 'sql' });

  const { updateInstanceList, generateInstanceSelectOption } = useInstance();
  const [instanceName, setInstanceName] = useState<string | undefined>(
    undefined
  );
  const { generateInstanceSchemaSelectOption } =
    useInstanceSchema(instanceName);
  const [maxQueryRows, setMaxQueryRows] = useState(defaultMaxQueryRows);
  const [
    showHistoryModal,
    { setFalse: closeHistoryModal, setTrue: openHistoryModal },
  ] = useBoolean(false);
  const [getSqlQueryLoading, { setFalse: finishQuery, setTrue: startQuery }] =
    useBoolean(false);

  const getSqlQueryResultList = async () => {
    const formValues = await form.validateFields();
    const prepareSqlQueryParams: IPrepareSQLQueryParams = {
      instance_name: formValues.instanceName,
      instance_schema: formValues.instanceSchema,
      sql: formValues.sql,
    };
    const prepareSqlQueryRes = await sql_query.prepareSQLQuery(
      prepareSqlQueryParams
    );
    if (prepareSqlQueryRes.data.code !== ResponseCode.SUCCESS) {
      setResultErrorMessage(
        prepareSqlQueryRes.data.message ?? t('common.unknownError')
      );
      return;
    }
    const actionArray = prepareSqlQueryRes.data.data?.query_ids?.map(
      ({ query_id }) => {
        const getSqlResultParams: IGetSQLResultParams = {
          query_id: query_id ?? '',
          page_size: formValues.maxPreQueryRows,
          page_index: 1,
        };
        return sql_query.getSQLResult(getSqlResultParams);
      }
    );
    const sqlQueryResultList = await Promise.all(actionArray ?? []);
    const failQuery = sqlQueryResultList.find(
      (v) => v.data.code !== ResponseCode.SUCCESS
    );
    if (failQuery) {
      setResultErrorMessage(failQuery.data.message ?? t('common.unknownError'));
      return;
    }

    const realQueryRes: SqlQueryResultType[] =
      prepareSqlQueryRes.data.data?.query_ids?.map((v, index) => {
        return {
          sqlQueryId: v.query_id ?? '',
          resultItem: sqlQueryResultList[index].data.data ?? {},
          hide: false,
        };
      }) ?? [];
    setQueryRes(realQueryRes);
  };

  const submit = () => {
    startQuery();
    getSqlQueryResultList().finally(() => {
      finishQuery();
    });
  };

  const applyHistorySql = useCallback(
    (sql: string) => {
      form.setFieldsValue({
        sql,
      });
      closeHistoryModal();
    },
    [closeHistoryModal, form]
  );

  useEffect(() => {
    updateInstanceList();
  }, [updateInstanceList]);

  useEffect(() => {
    if (instanceName) {
      instance.getInstanceV1({ instance_name: instanceName }).then((res) => {
        if (res) {
          const rowsLength =
            res.data.data?.sql_query_config?.max_pre_query_rows ??
            defaultMaxQueryRows;
          form.setFieldsValue({
            maxPreQueryRows:
              rowsLength > defaultMaxQueryRows
                ? defaultMaxQueryRows
                : rowsLength,
          });
          setMaxQueryRows(rowsLength);
        }
      });
    }
  }, [form, instanceName]);

  return (
    <>
      <Card title={t('sqlQuery.sqlInput.title')}>
        <Form<ISqlInputForm> onFinish={submit} form={form}>
          <Row {...FilterFormRowLayout}>
            <Col {...FilterFormColLayout}>
              <Form.Item
                name="instanceName"
                label={t('sqlQuery.sqlInput.instance')}
                rules={[
                  {
                    required: true,
                  },
                ]}
              >
                <Select
                  data-testid="instance-name"
                  value={instanceName}
                  onChange={setInstanceName}
                  placeholder={t('common.form.placeholder.select')}
                  className="middle-select"
                  allowClear
                  showSearch
                >
                  {generateInstanceSelectOption()}
                </Select>
              </Form.Item>
            </Col>
            <Col {...FilterFormColLayout}>
              <Form.Item
                name="instanceSchema"
                label={t('sqlQuery.sqlInput.database')}
                rules={[
                  {
                    required: true,
                  },
                ]}
              >
                <Select
                  placeholder={t('common.form.placeholder.select')}
                  showSearch
                  allowClear
                >
                  {generateInstanceSchemaSelectOption()}
                </Select>
              </Form.Item>
            </Col>
          </Row>
          <Form.Item
            name="sql"
            label=""
            initialValue="/* input your sql */"
            wrapperCol={{
              className: theme.editor,
            }}
            rules={[
              {
                required: true,
              },
            ]}
          >
            <MonacoEditor
              theme={currentEditorTheme}
              width="100%"
              height="200"
              language="sql"
              editorDidMount={editorDidMount}
            />
          </Form.Item>
          <Space size="large">
            <Form.Item
              label={t('sqlQuery.sqlInput.returnLength')}
              name="maxPreQueryRows"
              style={{ marginBottom: 0 }}
              initialValue={defaultMaxQueryRows}
              rules={[
                {
                  required: true,
                },
              ]}
            >
              <InputNumber max={maxQueryRows} />
            </Form.Item>

            <Form.Item label="" colon={false} style={{ marginBottom: 0 }}>
              <Button
                htmlType="submit"
                type="primary"
                loading={getSqlQueryLoading}
              >
                {t('sqlQuery.sqlInput.searchSqlResult')}
              </Button>
            </Form.Item>

            <Button
              disabled={!instanceName}
              type="primary"
              onClick={openHistoryModal}
            >
              {t('sqlQuery.sqlInput.sqlHistory')}
            </Button>
          </Space>
        </Form>
      </Card>
      <SqlQueryHistory
        visible={showHistoryModal}
        close={closeHistoryModal}
        applyHistorySql={applyHistorySql}
        instanceName={instanceName ?? ''}
      />
    </>
  );
};

export default SqlInput;
