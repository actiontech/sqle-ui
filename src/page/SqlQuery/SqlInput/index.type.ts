import { FormInstance } from 'antd';
import { ISqlInputForm } from '../index.type';

export interface SqlInputProps {
  form: FormInstance<ISqlInputForm>;
  dataSourceName: string;
  maxQueryRows?: number;
  updateDataSourceName: (name: string) => void;
  updateSchemaName: (name: string) => void;
  submitForm: (values: ISqlInputForm) => Promise<void>;
  getSQLExecPlan: () => Promise<void>;
}
