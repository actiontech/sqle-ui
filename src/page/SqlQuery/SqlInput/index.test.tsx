import { ThemeProvider } from '@material-ui/styles';
import { fireEvent, screen, waitFor } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import { useForm } from 'antd/lib/form/Form';
import SqlInput from '.';
import instance from '../../../api/instance';
import { renderWithRedux } from '../../../testUtils/customRender';
import { mockUseSelector } from '../../../testUtils/mockRedux';
import {
  mockUseInstance,
  mockUseInstanceSchema,
  resolveErrorThreeSecond,
  resolveThreeSecond,
} from '../../../testUtils/mockRequest';
import { SupportTheme } from '../../../theme';
import lightTheme from '../../../theme/light';
import { instance_info, query_ids_test1, query_res_1 } from '../__testData__';
import sql_query from '../../../api/sql_query';

describe('sqlQuery/sqlInput', () => {
  let getInstanceSpy: jest.SpyInstance;
  let getInstanceSchema: jest.SpyInstance;
  beforeEach(() => {
    jest.useFakeTimers();
    mockUseSelector({ user: { theme: SupportTheme.LIGHT } });
    getInstanceSpy = mockUseInstance();
    getInstanceSchema = mockUseInstanceSchema();
  });
  afterEach(() => {
    jest.useRealTimers();
    jest.clearAllMocks();
    jest.clearAllTimers();
  });

  const mockGetInstanceByName = () => {
    const spy = jest.spyOn(instance, 'getInstanceV1');
    spy.mockImplementation(() => {
      return resolveThreeSecond(instance_info);
    });
    return spy;
  };

  const mockPrePareSqlQuery = () => {
    const spy = jest.spyOn(sql_query, 'prepareSQLQuery');
    spy.mockImplementation(() => {
      return resolveThreeSecond(query_ids_test1);
    });
    return spy;
  };
  const mockGetSqlResult = () => {
    const spy = jest.spyOn(sql_query, 'getSQLResult');
    spy.mockImplementation(() => {
      return resolveThreeSecond(query_res_1);
    });
    return spy;
  };

  const mockPrePareSqlQueryHasError = () => {
    const spy = jest.spyOn(sql_query, 'prepareSQLQuery');
    spy.mockImplementation(() => {
      return resolveErrorThreeSecond({});
    });
    return spy;
  };

  const mockGetSqlResultHasError = () => {
    const spy = jest.spyOn(sql_query, 'getSQLResult');
    spy.mockImplementation(() => {
      return resolveErrorThreeSecond({});
    });
    return spy;
  };

  test('should get the instance list and match the snapshot', () => {
    const { result } = renderHook(() => useForm());
    const setQueryRes = jest.fn();
    const setResultErrorMessage = jest.fn();
    const { baseElement } = renderWithRedux(
      <ThemeProvider theme={lightTheme}>
        <SqlInput
          form={result.current[0]}
          setQueryRes={setQueryRes}
          setResultErrorMessage={setResultErrorMessage}
        />
      </ThemeProvider>
    );
    expect(setQueryRes).toBeCalledTimes(0);
    expect(setResultErrorMessage).toBeCalledTimes(0);
    expect(getInstanceSpy).toBeCalledTimes(1);
    expect(baseElement).toMatchSnapshot();
  });

  test('should disabled open history modal button when instance is empty', async () => {
    const { result } = renderHook(() => useForm());
    const setQueryRes = jest.fn();
    const setResultErrorMessage = jest.fn();
    mockGetInstanceByName();
    renderWithRedux(
      <ThemeProvider theme={lightTheme}>
        <SqlInput
          form={result.current[0]}
          setQueryRes={setQueryRes}
          setResultErrorMessage={setResultErrorMessage}
        />
      </ThemeProvider>
    );
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    expect(
      screen.getByText('sqlQuery.sqlInput.sqlHistory').parentNode
    ).toHaveAttribute('disabled');
    fireEvent.mouseDown(screen.getByLabelText('sqlQuery.sqlInput.instance'));
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });
    const instanceOptions = screen.getAllByText('instance1');
    const instance = instanceOptions[1];
    fireEvent.click(instance);
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    expect(
      screen.getByText('sqlQuery.sqlInput.sqlHistory').parentNode
    ).not.toHaveAttribute('disabled');
  });

  test('should be get instanceSchema list and set maxPreQueryRows value when select instanceName', async () => {
    const { result } = renderHook(() => useForm());
    const setQueryRes = jest.fn();
    const setResultErrorMessage = jest.fn();
    const getInstanceByName = mockGetInstanceByName();
    renderWithRedux(
      <ThemeProvider theme={lightTheme}>
        <SqlInput
          form={result.current[0]}
          setQueryRes={setQueryRes}
          setResultErrorMessage={setResultErrorMessage}
        />
      </ThemeProvider>
    );
    expect(getInstanceByName).toBeCalledTimes(0);

    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    fireEvent.mouseDown(screen.getByLabelText('sqlQuery.sqlInput.instance'));
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });
    const instanceOptions = screen.getAllByText('instance1');
    const instance = instanceOptions[1];
    expect(instance).toHaveClass('ant-select-item-option-content');
    fireEvent.click(instance);
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    expect(getInstanceByName).toBeCalledTimes(1);
    expect(getInstanceSchema).toBeCalledTimes(1);
    expect(screen.getByLabelText('sqlQuery.sqlInput.returnLength')).toHaveValue(
      instance_info.sql_query_config?.max_pre_query_rows!.toString()
    );
  });

  test('should execute getSqlQueryResultList when clicking submit', async () => {
    const { result } = renderHook(() => useForm());
    const setQueryRes = jest.fn();
    const setResultErrorMessage = jest.fn();
    mockGetInstanceByName();
    const prePareSqlQuery = mockPrePareSqlQuery();
    const getSqlResult = mockGetSqlResult();

    renderWithRedux(
      <ThemeProvider theme={lightTheme}>
        <SqlInput
          form={result.current[0]}
          setQueryRes={setQueryRes}
          setResultErrorMessage={setResultErrorMessage}
        />
      </ThemeProvider>
    );
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    fireEvent.mouseDown(screen.getByLabelText('sqlQuery.sqlInput.instance'));
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });
    const instanceOptions = screen.getAllByText('instance1');
    const instance = instanceOptions[1];
    fireEvent.click(instance);
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    fireEvent.mouseDown(screen.getByLabelText('sqlQuery.sqlInput.database'));
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });
    const schemaOptions = screen.getAllByText('schema1');
    fireEvent.click(schemaOptions[1]);

    fireEvent.click(screen.getByText('sqlQuery.sqlInput.searchSqlResult'));
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });
    expect(prePareSqlQuery).toBeCalledTimes(1);
    expect(
      screen.getByText('sqlQuery.sqlInput.searchSqlResult').parentNode
    ).toHaveClass('ant-btn-loading');
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    expect(getSqlResult).toBeCalledTimes(query_ids_test1.query_ids?.length!);
    expect(setQueryRes).toBeCalledTimes(1);
    expect(setResultErrorMessage).toBeCalledTimes(1);
    expect(
      screen.getByText('sqlQuery.sqlInput.searchSqlResult').parentNode
    ).not.toHaveClass('ant-btn-loading');
  });

  test('should show error message when prepare sql has error', async () => {
    const { result } = renderHook(() => useForm());
    const setQueryRes = jest.fn();
    const setResultErrorMessage = jest.fn();
    mockGetInstanceByName();
    const prePareSqlQueryHasError = mockPrePareSqlQueryHasError();
    const getSqlResult = mockGetSqlResult();

    renderWithRedux(
      <ThemeProvider theme={lightTheme}>
        <SqlInput
          form={result.current[0]}
          setQueryRes={setQueryRes}
          setResultErrorMessage={setResultErrorMessage}
        />
      </ThemeProvider>
    );
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    fireEvent.mouseDown(screen.getByLabelText('sqlQuery.sqlInput.instance'));
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });
    const instanceOptions = screen.getAllByText('instance1');
    const instance = instanceOptions[1];
    fireEvent.click(instance);
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    fireEvent.mouseDown(screen.getByLabelText('sqlQuery.sqlInput.database'));
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });
    const schemaOptions = screen.getAllByText('schema1');
    fireEvent.click(schemaOptions[1]);

    fireEvent.click(screen.getByText('sqlQuery.sqlInput.searchSqlResult'));
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });

    expect(prePareSqlQueryHasError).toBeCalledTimes(1);
    expect(
      screen.getByText('sqlQuery.sqlInput.searchSqlResult').parentNode
    ).toHaveClass('ant-btn-loading');
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    expect(setResultErrorMessage).toBeCalledTimes(1);
    expect(setQueryRes).toBeCalledTimes(0);
    expect(getSqlResult).toBeCalledTimes(0);
    expect(
      screen.getByText('sqlQuery.sqlInput.searchSqlResult').parentNode
    ).not.toHaveClass('ant-btn-loading');
  });
  test('should show error message when get sql result has error', async () => {
    const { result } = renderHook(() => useForm());
    const setQueryRes = jest.fn();
    const setResultErrorMessage = jest.fn();
    mockGetInstanceByName();
    const prePareSqlQuery = mockPrePareSqlQuery();
    const getSqlResultHasError = mockGetSqlResultHasError();

    renderWithRedux(
      <ThemeProvider theme={lightTheme}>
        <SqlInput
          form={result.current[0]}
          setQueryRes={setQueryRes}
          setResultErrorMessage={setResultErrorMessage}
        />
      </ThemeProvider>
    );
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    fireEvent.mouseDown(screen.getByLabelText('sqlQuery.sqlInput.instance'));
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });
    const instanceOptions = screen.getAllByText('instance1');
    const instance = instanceOptions[1];
    fireEvent.click(instance);
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    fireEvent.mouseDown(screen.getByLabelText('sqlQuery.sqlInput.database'));
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });
    const schemaOptions = screen.getAllByText('schema1');
    fireEvent.click(schemaOptions[1]);

    fireEvent.click(screen.getByText('sqlQuery.sqlInput.searchSqlResult'));
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });

    expect(prePareSqlQuery).toBeCalledTimes(1);
    expect(
      screen.getByText('sqlQuery.sqlInput.searchSqlResult').parentNode
    ).toHaveClass('ant-btn-loading');
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    expect(getSqlResultHasError).toBeCalledTimes(
      query_ids_test1.query_ids?.length!
    );
    expect(setResultErrorMessage).toBeCalledTimes(1);
    expect(setQueryRes).toBeCalledTimes(0);
    expect(
      screen.getByText('sqlQuery.sqlInput.searchSqlResult').parentNode
    ).not.toHaveClass('ant-btn-loading');
  });
});
