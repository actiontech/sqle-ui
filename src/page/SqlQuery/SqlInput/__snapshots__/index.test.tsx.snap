// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sqlQuery/sqlInput should get the instance list and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="ant-card ant-card-bordered"
    >
      <div
        class="ant-card-head"
      >
        <div
          class="ant-card-head-wrapper"
        >
          <div
            class="ant-card-head-title"
          >
            sqlQuery.sqlInput.title
          </div>
        </div>
      </div>
      <div
        class="ant-card-body"
      >
        <form
          class="ant-form ant-form-horizontal"
        >
          <div
            class="ant-row"
            style="margin-left: -12px; margin-right: -12px;"
          >
            <div
              class="ant-col ant-col-xs-24 ant-col-sm-12 ant-col-xl-8 ant-col-xxl-6"
              style="padding-left: 12px; padding-right: 12px;"
            >
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-col ant-form-item-label"
                >
                  <label
                    class="ant-form-item-required"
                    for="instanceName"
                    title="sqlQuery.sqlInput.instance"
                  >
                    sqlQuery.sqlInput.instance
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <div
                        class="ant-select middle-select ant-select-single ant-select-allow-clear ant-select-show-arrow ant-select-show-search"
                        data-testid="instance-name"
                      >
                        <div
                          class="ant-select-selector"
                        >
                          <span
                            class="ant-select-selection-search"
                          >
                            <input
                              aria-activedescendant="instanceName_list_0"
                              aria-autocomplete="list"
                              aria-controls="instanceName_list"
                              aria-haspopup="listbox"
                              aria-owns="instanceName_list"
                              autocomplete="off"
                              class="ant-select-selection-search-input"
                              id="instanceName"
                              role="combobox"
                              type="search"
                              value=""
                            />
                          </span>
                          <span
                            class="ant-select-selection-placeholder"
                          >
                            common.form.placeholder.select
                          </span>
                        </div>
                        <span
                          aria-hidden="true"
                          class="ant-select-arrow"
                          style="user-select: none;"
                          unselectable="on"
                        >
                          <span
                            aria-label="down"
                            class="anticon anticon-down ant-select-suffix"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="down"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                              />
                            </svg>
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-col ant-col-xs-24 ant-col-sm-12 ant-col-xl-8 ant-col-xxl-6"
              style="padding-left: 12px; padding-right: 12px;"
            >
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-col ant-form-item-label"
                >
                  <label
                    class="ant-form-item-required"
                    for="instanceSchema"
                    title="sqlQuery.sqlInput.database"
                  >
                    sqlQuery.sqlInput.database
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <div
                        class="ant-select ant-select-single ant-select-allow-clear ant-select-show-arrow ant-select-show-search"
                      >
                        <div
                          class="ant-select-selector"
                        >
                          <span
                            class="ant-select-selection-search"
                          >
                            <input
                              aria-activedescendant="instanceSchema_list_0"
                              aria-autocomplete="list"
                              aria-controls="instanceSchema_list"
                              aria-haspopup="listbox"
                              aria-owns="instanceSchema_list"
                              autocomplete="off"
                              class="ant-select-selection-search-input"
                              id="instanceSchema"
                              role="combobox"
                              type="search"
                              value=""
                            />
                          </span>
                          <span
                            class="ant-select-selection-placeholder"
                          >
                            common.form.placeholder.select
                          </span>
                        </div>
                        <span
                          aria-hidden="true"
                          class="ant-select-arrow"
                          style="user-select: none;"
                          unselectable="on"
                        >
                          <span
                            aria-label="down"
                            class="anticon anticon-down ant-select-suffix"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="down"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                              />
                            </svg>
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control makeStyles-editor-3"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    height="200"
                    id="sql"
                    language="sql"
                    theme="vs"
                    value="/* input your sql */"
                    width="100%"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-space ant-space-horizontal ant-space-align-center"
          >
            <div
              class="ant-space-item"
              style="margin-right: 24px;"
            >
              <div
                class="ant-row ant-form-item"
                style="margin-bottom: 0px;"
              >
                <div
                  class="ant-col ant-form-item-label"
                >
                  <label
                    class="ant-form-item-required"
                    for="maxPreQueryRows"
                    title=""
                  >
                    <div
                      class="ant-space ant-space-horizontal ant-space-align-center"
                    >
                      <div
                        class="ant-space-item"
                        style="margin-right: 8px;"
                      >
                        <span
                          aria-label="info-circle"
                          class="anticon anticon-info-circle text-orange"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="info-circle"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                            />
                            <path
                              d="M464 336a48 48 0 1096 0 48 48 0 10-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
                            />
                          </svg>
                        </span>
                      </div>
                      <div
                        class="ant-space-item"
                      >
                        sqlQuery.sqlInput.returnLength
                      </div>
                    </div>
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <div
                        class="ant-input-number"
                      >
                        <div
                          class="ant-input-number-handler-wrap"
                        >
                          <span
                            aria-disabled="true"
                            aria-label="Increase Value"
                            class="ant-input-number-handler ant-input-number-handler-up ant-input-number-handler-up-disabled"
                            role="button"
                            unselectable="on"
                          >
                            <span
                              aria-label="up"
                              class="anticon anticon-up ant-input-number-handler-up-inner"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="up"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                                />
                              </svg>
                            </span>
                          </span>
                          <span
                            aria-disabled="false"
                            aria-label="Decrease Value"
                            class="ant-input-number-handler ant-input-number-handler-down"
                            role="button"
                            unselectable="on"
                          >
                            <span
                              aria-label="down"
                              class="anticon anticon-down ant-input-number-handler-down-inner"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="down"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                />
                              </svg>
                            </span>
                          </span>
                        </div>
                        <div
                          class="ant-input-number-input-wrap"
                        >
                          <input
                            aria-valuemax="100"
                            aria-valuenow="100"
                            autocomplete="off"
                            class="ant-input-number-input"
                            id="maxPreQueryRows"
                            role="spinbutton"
                            step="1"
                            value="100"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-space-item"
              style="margin-right: 24px;"
            >
              <div
                class="ant-row ant-form-item"
                style="margin-bottom: 0px;"
              >
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <button
                        class="ant-btn ant-btn-primary"
                        type="submit"
                      >
                        <span>
                          sqlQuery.sqlInput.searchSqlResult
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-space-item"
            >
              <button
                class="ant-btn ant-btn-primary"
                disabled=""
                type="button"
              >
                <span>
                  sqlQuery.sqlInput.sqlHistory
                </span>
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</body>
`;
